ext/calendar/gregor.c

```diff
@@ -160,6 +160,10 @@ void SdnToGregorian(
 	/* Calculate the century (year/100). */
 	century = temp / DAYS_PER_400_YEARS;
 
+	if (century > ((INT_MAX / 100) - (temp / DAYS_PER_4_YEARS))) {
```

Comment:
Shouldn't  this check be after the reassignment of `temp`? `temp` itself can't overflow I think?

Reply:
true I missed that line.